libcaf_core/caf/json_value.test.cpp

```diff
@@ -367,3 +367,75 @@ TEST("non-empty object with nested values") {
   check_eq(obj.value("min-uint64").to_unsigned(), 0u);
   check_eq(obj.value("max-uint64").to_unsigned(), UINT64_MAX);
 }
+
+TEST("non-empty object with null or nan values") {
+  std::string_view json_with_nan_or_null_values = R"_({
+  "null-value": null,
+  "nan-value": nan
+})_";
```

Comment:
Hard to read since it breaks the indentation:

```suggestion
    "null-value": null,
    "nan-value": nan
  })_";
```

Same for the other tests.